### PR TITLE
#963: open externally read from context features

### PR DESCRIPTION
**Pull request by joao-conde**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-white/issues/963 |
| Dependencies | -- |
| Decisions | Receive open externally prop. |

### CODE REVIEW

File: vue/components/forms/personalization-form/initials-images.vue [hunk 1]
```diff
@@ -215,6 +215,14 @@ export const InitialsImages = {
             if (this.imageBorderRadius) base["border-radius"] = `${this.imageBorderRadius}`;
             if (this.imageObjectFit) base["object-fit"] = this.imageObjectFit;
             return base;
+        },
+        _openExternally() {
```
Comment (BeeMargarida) [inline vue/components/forms/personalization-form/initials-images.vue:219]: we normally don't use the private prefix `_` in computed, so I think you can rename it to `open` or something else

File: vue/components/forms/personalization-form/initials-images.vue [hunk 2]
```diff
@@ -215,6 +215,14 @@ export const InitialsImages = {
...
             if (this.imageObjectFit) base["object-fit"] = this.imageObjectFit;
             return base;
+        },
+        _openExternally() {
+            if (this.openExternally !== null) return this.openExternally;
+
+            const features = this.$store.state.features;
+            if (features["open-externally"] !== undefined) return features["open-externally"];
+
+            return false;
```
Comment (BeeMargarida) [inline vue/components/forms/personalization-form/initials-images.vue:225]: can't you simplify with something like
```suggestion
            const features = this.$store.state.features || {};
            return features["open-externally"] !== undefined ? features["open-externally"] : this.openExternally; 
```

File: vue/components/forms/personalization-form/initials-images.vue [hunk 3]
```diff
@@ -160,7 +160,7 @@ export const InitialsImages = {
          */
         openExternally: {
             type: Boolean,
-            default: false
+            default: null
```
Comment (BeeMargarida) [inline vue/components/forms/personalization-form/initials-images.vue:163]: Why did you change this?

File: vue/components/forms/personalization-form/initials-images.vue (hunk 2 above)
Comment (joao-conde) [reply vue/components/forms/personalization-form/initials-images.vue:225]: In this case, what if neither the feature nor the prop is defined? You get null instead of false.

File: vue/components/forms/personalization-form/initials-images.vue (hunk 3 above)
Comment (joao-conde) [reply vue/components/forms/personalization-form/initials-images.vue:163]: I wanted to give priority to the prop, then the feature, then default to false.

If you default the prop to false, the feature would never be used.

File: vue/components/forms/personalization-form/initials-images.vue (hunk 2 above)
Comment (BeeMargarida) [reply vue/components/forms/personalization-form/initials-images.vue:225]: The boolean value of null is false, so it's not a problem. Either way, why change the default value of the prop to null instead of false?

File: vue/components/forms/personalization-form/initials-images.vue (hunk 3 above)
Comment (BeeMargarida) [reply vue/components/forms/personalization-form/initials-images.vue:163]: Got it!

File: vue/components/forms/personalization-form/initials-images.vue (hunk 2 above)
Comment (joao-conde) [reply vue/components/forms/personalization-form/initials-images.vue:225]: Because if it is false, I can not know if the prop was given or defaulted, and hence I cant use the logic of first the prop, then the feature, then false default.

Comment (BeeMargarida) [reply vue/components/forms/personalization-form/initials-images.vue:225]: I forgot you wanted to give priority to the prop, perhaps something like:
```suggestion
            const features = this.$store.state.features || {};
            if (this.openExternally !== null) return this.openExternally;
            return features["open-externally"] !== undefined ? features["open-externally"] : this.openExternally; 
```

Comment (joao-conde) [reply vue/components/forms/personalization-form/initials-images.vue:225]: This would return null if no prop or feature is given. I know in javascript it will have the same behavior, but I really think we should return false for a boolean property.

Comment (BeeMargarida) [reply vue/components/forms/personalization-form/initials-images.vue:225]: Sure

**Review by BeeMargarida — APPROVED** [review]

**Review by hugo-gomes — APPROVED** [review]

File: vue/components/forms/personalization-form/initials-images.vue [hunk 4]
```diff
@@ -215,6 +215,12 @@ export const InitialsImages = {
             if (this.imageBorderRadius) base["border-radius"] = `${this.imageBorderRadius}`;
             if (this.imageObjectFit) base["object-fit"] = this.imageObjectFit;
             return base;
+        },
+        openExternallyFeature() {
+            const features = this.$store.state.features || {};
```
Comment (joamag) [inline vue/components/forms/personalization-form/initials-images.vue:220]: We should avoid this strategy as the "features" concept is a White idea and not part of the ripe-commons domain. I believe we'll have to do this completely inside white codebase.

**Review by joamag — CHANGES_REQUESTED** [review]

**Comment by joamag** [comment]
@joao-conde I believe we'll have to somehow move the solution into white's codebase

File: vue/components/forms/personalization-form/initials-images.vue (hunk 4 above)
Comment (joao-conde) [reply vue/components/forms/personalization-form/initials-images.vue:220]: I had that idea initially and changed the clients of this component to pass the correct prop value. But that implies changing white where it uses this component as well as all retail vendors and plugins.

It does make more sense because the component shouldn't know about the concept of features.

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-*
+* Receive `openExternally` prop on the `personalization` component - [#963](https://github.com/ripe-tech/ripe-white/issues/963)
```
Comment (BeeMargarida) [inline CHANGELOG.md:12]:
```suggestion
* Receive `openExternally` prop on the `personalization` component - [ripe-white/#963](https://github.com/ripe-tech/ripe-white/issues/963)
```

File: vue/components/organisms/personalization/personalization.vue [hunk 6]
```diff
@@ -23,6 +23,7 @@
                     {{ modalSubTitleComputed }}
                 </h4>
                 <component
+                    v-bind="$props"
```
Comment (BeeMargarida) [inline vue/components/organisms/personalization/personalization.vue:26]: this is dangerous, we do not want to pass all props, only openExternally

**Review by BeeMargarida — APPROVED** [review]

**Review by hugo-gomes — APPROVED** [review]

**Review by joamag — APPROVED** [review]
